myhoard/controller.py

```diff
@@ -1078,6 +1078,15 @@ def _purge_old_backups(self):
             owned_stream_ids = [sid for sid in self.state["owned_stream_ids"] if sid != backup["stream_id"]]
             self.state_manager.update_state(owned_stream_ids=owned_stream_ids)
 
+    def _get_oldest_binlog_time(self):
```

Comment:
nitpicking: it could be written as `sorted(lambda b: b['processed_at'])[-1]`

Reply:
Would need something a bit more complex for handling empty binlogs without index error but this actually reminds me that the binlogs are already always in sorted order. The first in the list is the oldest. So looping through the whole list is actually just wasting CPU cycles.

Reply:
@derrix060 I disagree having a function makes it clear what it is doing
@rikonen  premature optimization is the root of all evil =)

Reply:
Only when what is done is a bit complex.  You wouldn't be creating a function called `sum` just to add two numbers for example, or create a function called `elapsed_time` just to do what you did in the other file: `current_time - last_could_have_purged`

Reply:
your solution takes way longer to read and understand opposed to a function which name says exactly what it does

Reply:
Well I don't know if it's really premature optimization or just relying on expected behavior. Other code in the same class already makes that assumption:

```
        for binlog in binlogs:
            binlog_age = time.time() - binlog["processed_at"]
            min_age = purge_settings["min_binlog_age_before_purge"]
            if binlog_age < min_age:
                log.info(
                    "Binlog %s was processed %s seconds ago and min age before purging is %s seconds, not purging",
                    binlog["local_index"], math.ceil(binlog_age), min_age
                )
                break
```

And I'm not actually worried about the lost CPU cycles but rather code being confusing because it does something that seemingly shouldn't be needed and then one is left wondering which part is making assumptions it shouldn't.

But anyway can be left as it is too if you want, doesn't matter much in the end and there's already been quite a bit of iteration for this. (I could've read this properly from get go, have just looked into this briefly between handling escalations.)